src/ai/backend/agent/stats.py

```diff
@@ -65,18 +68,39 @@ def check_cgroup_available():
 
 
 class StatModes(enum.Enum):
+    """
+    CGROUP = Tries to automatically detect docker's current cgroup driver
+    CGROUPFS = Manually set cgroup driver to cgroupfs
+    SYSTEMD = Manually set cgroup driver to systemd
+    """
+
     CGROUP = "cgroup"
+    CGROUPFS = "cgroupfs"
+    SYSTEMD = "systemd"
     DOCKER = "docker"
 
     @staticmethod
-    def get_preferred_mode():
+    async def get_preferred_mode():
         """
         Returns the most preferred statistics collector type for the host OS.
         """
         if check_cgroup_available():
-            return StatModes.CGROUP
+            docker_cgroup_driver = await StatModes.check_docker_cgroup_driver()
+            if docker_cgroup_driver is not None:
+                return docker_cgroup_driver
+            else:  # Fallback to default (cgroupfs)
+                return StatModes.CGROUPFS
         return StatModes.DOCKER
 
+    @classmethod
+    async def check_docker_cgroup_driver(cls):
```

Comment:
Could you specify the return type explicitly?
If required, you may add `from __future__ import annotations`.